tests/drivers/adc/adc_api/src/test_adc.c

```diff
@@ -125,6 +125,9 @@ static int test_task_multiple_channels(void)
 	}
 
 	ret = adc_read(adc_channels[0].dev, &sequence);
+	if (ret == -ENOTSUP) {
```

Comment:
Is this really needed? Maybe it is better to remove the unsupported target from the yaml.

Reply:
The problem I ran into is that only some of the tests should be skipped, not all of them.  Splitting the test case into two different tests could be a solution.  I'll take a look over the next day or two and see if there is anything better.

Reply:
What tests should be skipped and for what reasons?

Reply:
The adc_esp32.c implementation does not support extra samples, interval sampling, and multi-channel sampling so it reports -ENOTSUP in those cases.

Previously, the adc_api test was not being run, so once I enabled the test, it failed when trying to test unsupported features in adc_esp32.c.

Really, we have three options here:
1. Split the unit test into separate tests for different modes
2. Update adc_esp32.c to add the missing features
3. Do the skip tests like I did here

Reply:
I had another look at this and I do not see any better approach.  The ADC API test tests the mandatory ADC API calls of channel_setup and read which drivers/adc/adc_esp32.c does support.

https://github.com/zephyrproject-rtos/zephyr/blob/b92e04e636b5273af02e441f65d9eacedb91d84e/include/zephyr/drivers/adc.h#L593-L605

In addition, the test will verify option features such as multiple channels, and sequential reads which this driver does not support.  There is no API and there is no CONFIG_ option to disable these, so the unit test just needs to skip the test case if it isn't supported.

> Is this really needed? Maybe it is better to remove the unsupported target from the yaml.

If the unit test were split into mandatory and option, this would be the correct approach since nothing would need to be skipped.  However, the unit test is not split into multiple test cases, so we would need to split it into multiple parts and then configure each of the 183 boards that use this unit tests.  This seems like a lot of work with no real gain.

Let me know if I am missing something that would change this opinion.

Reply:
See previous reply.